### PR TITLE
expression: Add missing pbcode for functions `InetAton/InetNtoa/Inet6Aton/Inet6Ntoa/IsIPv4/IsIPv4Compat/IsIPv4Mapped/IsIPv6`.

### PR DESCRIPTION
**Pull request by LittleFall**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/26942 

Problem Summary: some scalar function hasn't pdcode for some reason, so it can't be pushed down as expected.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed: 
1. add warnings for functions without pbcode.
2. add pbcode for some functions.
2. ban push down to tikv for these functions.
4. add tests for them.

How it Works:

### Check List 

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by fuzhe1989 — APPROVED** [review]

File: expression/expr_to_pb_test.go [hunk 1]
```diff
@@ -1008,6 +1008,58 @@ func (s *testEvaluatorSuite) TestExprOnlyPushDownToFlash(c *C) {
...
+func (s *testEvaluatorSuite) TestExprPushDownToTiKV(c *C) {
+	sc := new(stmtctx.StatementContext)
+	client := new(mock.Client)
+	dg := new(dataGen4Expr2PbTest)
+	exprs := make([]Expression, 0)
+
+	//jsonColumn := dg.genColumn(mysql.TypeJSON, 1)
+	//intColumn := dg.genColumn(mysql.TypeLonglong, 2)
+	//realColumn := dg.genColumn(mysql.TypeDouble, 3)
+	//decimalColumn := dg.genColumn(mysql.TypeNewDecimal, 4)
```
Comment (ichn-hu) [inline expression/expr_to_pb_test.go:1020]: ?

Comment (LittleFall) [reply expression/expr_to_pb_test.go:1020]: later will use

**Review by hanfei1991 — APPROVED** [review]

File: expression/expression.go [hunk 2]
```diff
@@ -979,9 +979,10 @@ func scalarExprSupportedByTiKV(sf *ScalarFunction) bool {
 		ast.MD5, ast.SHA1, ast.UncompressedLength,
 
 		ast.Cast,
-
+
```
Comment (lzmhhh123) [inline expression/expression.go:982]: remove the tab.

**Review by lzmhhh123 — APPROVED** [review]

**Comment by LittleFall** [comment]
[2021-08-06T07:22:54.680Z] Using context: idc-jenkins-ci-tidb/check_dev

[2021-08-06T07:22:55.350Z] java.lang.InterruptedException

[2021-08-06T07:22:55.350Z] 	at java.lang.Object.wait(Native Method)

[2021-08-06T07:22:55.350Z] 	at java.lang.Object.wait(Object.java:502)

[2021-08-06T07:22:55.350Z] 	at java.lang.UNIXProcess.waitFor(UNIXProcess.java:395)

[2021-08-06T07:22:55.350Z] 	at hudson.Proc$LocalProc.join(Proc.java:327)

[2021-08-06T07:22:55.350Z] 	at hudson.Proc.joinWithTimeout(Proc.java:172)

[2021-08-06T07:22:55.350Z] 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2632)

[2021-08-06T07:22:55.350Z] 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:2079)

[2021-08-06T07:22:55.350Z] 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$500(CliGitAPIImpl.java:84)

[2021-08-06T07:22:55.350Z] 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:601)

[2021-08-06T07:22:55.350Z] 	at jenkins.plugins.git.GitSCMFileSystem$BuilderImpl.build(GitSCMFileSystem.java:365)

[2021-08-06T07:22:55.350Z] 	at jenkins.scm.api.SCMFileSystem.of(SCMFileSystem.java:197)

[2021-08-06T07:22:55.350Z] 	at jenkins.scm.api.SCMFileSystem.of(SCMFileSystem.java:173)

[2021-08-06T07:22:55.350Z] 	at org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition.create(CpsScmFlowDefinition.java:114)

[2021-08-06T07:22:55.350Z] 	at org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition.create(CpsScmFlowDefinition.java:68)

[2021-08-06T07:22:55.350Z] 	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:309)

[2021-08-06T07:22:55.350Z] 	at hudson.model.ResourceController.execute(ResourceController.java:97)

[2021-08-06T07:22:55.350Z] 	at hudson.model.Executor.run(Executor.java:429)

[2021-08-06T07:22:55.350Z] Finished: FAILURE

**Comment by zhouqiang-cl** [comment]
CI fails for many times
```

 FAIL: enforce_mpp_test.go:128: testEnforceMPPSuite.TestEnforceMPPWarning1

[2021-08-06T07:47:41.884Z] 

[2021-08-06T07:47:41.884Z] enforce_mpp_test.go:191:

[2021-08-06T07:47:41.884Z]     c.Assert(s.testData.ConvertSQLWarnToStrings(tk.Se.GetSessionVars().StmtCtx.GetWarnings()), DeepEquals, output[i].Warn)

[2021-08-06T07:47:41.884Z] ... obtained []string = []string{"Scalar function 'is_ipv4'(signature: IsIPv4) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs", "Scalar function 'is_ipv4'(signature: IsIPv4) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs", "Scalar function 'is_ipv4'(signature: IsIPv4) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs"}

[2021-08-06T07:47:41.884Z] ... expected []string = []string{"Scalar function 'is_ipv4'(signature: Unspecified) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs", "Scalar function 'is_ipv4'(signature: Unspecified) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs", "Scalar function 'is_ipv4'(signature: Unspecified) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs"}
```

in https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/23511/pipeline

**Comment by LittleFall** [comment]
> CI fails for many times
> 
> ```
> 
>  FAIL: enforce_mpp_test.go:128: testEnforceMPPSuite.TestEnforceMPPWarning1
> 
> [2021-08-06T07:47:41.884Z] 
> 
> [2021-08-06T07:47:41.884Z] enforce_mpp_test.go:191:
> 
> [2021-08-06T07:47:41.884Z]     c.Assert(s.testData.ConvertSQLWarnToStrings(tk.Se.GetSessionVars().StmtCtx.GetWarnings()), DeepEquals, output[i].Warn)
> 
> [2021-08-06T07:47:41.884Z] ... obtained []string = []string{"Scalar function 'is_ipv4'(signature: IsIPv4) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs", "Scalar function 'is_ipv4'(signature: IsIPv4) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs", "Scalar function 'is_ipv4'(signature: IsIPv4) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs"}
> 
> [2021-08-06T07:47:41.884Z] ... expected []string = []string{"Scalar function 'is_ipv4'(signature: Unspecified) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs", "Scalar function 'is_ipv4'(signature: Unspecified) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs", "Scalar function 'is_ipv4'(signature: Unspecified) can not be pushed to tiflash", "Aggregation can not be pushed to tiflash because groupByItems contain unsupported exprs"}
> ```
> 
> in https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/23511/pipeline

my code failed, fixed

**Comment by LittleFall** [comment]
[2021-08-06T08:02:11.933Z] + rm -rf /tmp/tidb

[2021-08-06T08:02:11.933Z] + mkdir -p plugin-so

[2021-08-06T08:02:11.933Z] + cp /home/jenkins/agent/workspace/tidb_ghpr_build/go/src/github.com/pingcap/enterprise-plugin/audit/audit-1.so ./plugin-so/

[2021-08-06T08:02:11.933Z] + cp /home/jenkins/agent/workspace/tidb_ghpr_build/go/src/github.com/pingcap/enterprise-plugin/whitelist/whitelist-1.so ./plugin-so/

[2021-08-06T08:02:12.192Z] + sleep 5

[2021-08-06T08:02:12.192Z] + /home/jenkins/agent/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/bin/tidb-server -plugin-dir=/home/jenkins/agent/workspace/tidb_ghpr_build/go/src/github.com/pingcap/tidb/plugin-so -plugin-load=audit-1,whitelist-1

[2021-08-06T08:02:17.453Z] + mysql -h 127.0.0.1 -P 4000 -u root -e 'select tidb_version()'

[2021-08-06T08:02:17.453Z] ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1' (111)

**Pull request merged**